### PR TITLE
transition-donation-status/id

### PR DESCRIPTION
**Pull request by Abdalrazaq94**
## transition-donation-status/id

when someone click on donate will go to different page

## Description

Write a summary of your change. Explain why you have made it.

## Related to

Make sure you include the issue number with a hash sign # so Github can link this PR to the right issue, like this:

Fixes #1

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have carefully reviewed my own code
- [ ] I have commented my code
- [ ] I have updated any documentation

### CODE REVIEW

File: client/src/pages/ViewRequestStatus/DonationRequestStatus.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+function DonationRequestStatues() {
+    const [request, setRequest] = useState(null);
+	// this helps get the id from the router
+	const { id } = useParams();
+
+    //  this use effect/ it may be not the right Api that we are going to Fetch the data with
+    // this only template 
+	useEffect(() => {
+		fetch(`/laptop_assignment/{id}`)
```
Comment (gregdyke) [inline client/src/pages/ViewRequestStatus/DonationRequestStatus.js]: it should be laptop_donation. But for now, this is not part of the ticket at all. You don't need the useEffect

File: package.json [hunk 2]
```diff
@@ -79,6 +79,7 @@
 		"express": "^4.18.1",
 		"helmet": "^6.0.0",
 		"morgan": "^1.10.0",
+		"navigate": "^0.3.6",
```
Comment (gregdyke) [inline package.json]: I suspect you haven't fully rebased on top of the main branch: navigate is already in package.json

File: server/db_setup.sql [hunk 3]
```diff
@@ -9,7 +9,7 @@ create table laptop_request (
     lastname varchar(100),
     email varchar(100),
     phonenumber varchar(100),
-    laptop_request_status laptop_request_status DEFAUlT 'ACTIVE'
+    laptop_request_status  DEFAUlT 'ACTIVE'
```
Comment (gregdyke) [inline server/db_setup.sql]: I think this is wrong?

**Review by gregdyke — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (gregdyke) [reply package.json]: Woops, got it wrong. You don't need navigate as you are not using it: you already have react-router-dom

Comment (Abdalrazaq94) [reply package.json]: fixed.
i downloaded it while i use testing the navigate

File: server/db_setup.sql (hunk 3 above)
Comment (Abdalrazaq94) [reply server/db_setup.sql]: fixed

Comment (gregdyke) [reply server/db_setup.sql]: I think it's still wrong. As best I can tell, `laptop_request_status laptop_request_status DEFAUlT 'ACTIVE'` is correct for that line

**Review by gregdyke — CHANGES_REQUESTED** [review]
I could be wrong about the sql - but Fareed's version seems to work for him, so I would assume it works for you too?

File: server/db_setup.sql (hunk 3 above)
Comment (Abdalrazaq94) [reply server/db_setup.sql]: I fixed back

**Review by gregdyke — APPROVED** [review]
Nice job!

**Comment by Abdalrazaq94** [comment]
@gregdyke the reason why i delete the laptop_request_status laptop_request_status DEFAUlT 'ACTIVE and just made it laptop request_status
as I thought by Default the first value in the type table is active and this will be always active

**Pull request merged**